File: backend/app/api/schemas/health.py
"""Schemas for GET /api/v1/health (SDD Section 10.7)."""

from typing import Literal

from pydantic import BaseModel

HealthStatus = Literal["healthy", "degraded", "unhealthy"]


class HealthChecks(BaseModel):
    api: str
    database: str
    vector_store: str
    embedding_model: str
    llm: str


class HealthResponse(BaseModel):
    status: HealthStatus
    checks: HealthChecks
    version: str

File: frontend/src/types/health.ts
// Mirrors backend/app/api/schemas/health.py

export type HealthStatus = 'healthy' | 'degraded' | 'unhealthy'

export interface HealthChecks {
  api: string
  database: string
  vector_store: string
  embedding_model: string
  llm: string
}

export interface HealthResponse {
  status: HealthStatus
  checks: HealthChecks
  version: string
}
